impl/ocaml/sqlgg_runtime.ml

```diff
@@ -1,10 +1,32 @@
 
-let replace_all_chars ~str ~sub ~by =
-  let rec get_chunks s sub acc off =
-    if off >= String.length s then List.rev ("" :: acc)
-    else
-      match String.index_from_opt s off sub with
-      | None -> List.rev (String.sub s off (String.length s - off) :: acc)
-      | Some off' -> get_chunks s sub (String.sub s off (off' - off) :: acc) (off' + 1)
-  in
-  String.concat by (get_chunks str sub [] 0)
+
+(* cf. https://dev.mysql.com/doc/refman/5.7/en/string-literals.html *)
+let mysql_string_to_literal s =
+  let b = Buffer.create (String.length s + String.length s lsl 2) in
+  Buffer.add_string b "'";
+  for i = 0 to String.length s - 1 do
+    match String.unsafe_get s i with
+    | '\\' -> Buffer.add_string b "\\\\"
+    | '\000' -> Buffer.add_string b "\\0"
+    | '\'' -> Buffer.add_string b "\\'"
+    | c -> Buffer.add_char b c
+  done;
+  Buffer.add_string b "'";
+  Buffer.contents b
+
+(* cf. https://sqlite.org/lang_expr.html "Literal Values"
+    "A string constant is formed by enclosing the string in single quotes
+    ('). A single quote within the string can be encoded by putting two
+    single quotes in a row - as in Pascal. C-style escapes using the
+    backslash character are not supported because they are not standard
+    SQL." *)
+let sqlite3_string_to_literal s =
+  let b = Buffer.create (String.length s + String.length s lsl 2) in
```

Comment:
> lsl 2

lets not go overboard, `* 4` is easier to read. But why such a huge buffer? it will never get fully used

Reply:
that was me trolling for the laughs with the bit shifting and shooting myself in the foot; changed to `len + len / 4`